app/models/form/custom_emoji_batch.rb

```diff
@@ -90,8 +90,8 @@ def copy!
     custom_emojis.each { |custom_emoji| authorize(custom_emoji, :copy?) }
 
     custom_emojis.each do |custom_emoji|
-      copied_custom_emoji = custom_emoji.copy!
-      log_action :create, copied_custom_emoji
+      custom_emoji.copy!
+      log_action :create, custom_emoji
```

Comment:
I need to check why output of `custom_emoji.copy!` doesn't work, but this would be wrong, since it will log the creation of the *remote* emoji, and not the local one...

Reply:
I understand your concern. It seems that the output of `custom_emoji.copy!` is just true/false. Following changes is acceptable for you?

```
diff --git a/app/models/custom_emoji.rb b/app/models/custom_emoji.rb
index edb1bec..0dacaf6 100644
--- a/app/models/custom_emoji.rb
+++ b/app/models/custom_emoji.rb
@@ -63,7 +63,7 @@ class CustomEmoji < ApplicationRecord
   def copy!
     copy = self.class.find_or_initialize_by(domain: nil, shortcode: shortcode)
     copy.image = image
-    copy.save!
+    copy.tap(&:save!)
   end
```

Reply:
Oh yes, that is a good fix! I've run into that mistake a few times before

Reply:
Sent new fix.